pkg/controller/master/storagenetwork/storage_network.go

```diff
@@ -0,0 +1,845 @@
+		if apierrors.IsNotFound(err) {
+			logrus.Infof("prometheus not found. skip")
+			return true
+		}
+		logrus.Warnf("prometheus get error %v", err)
+		return false
+	}
+
+	logrus.Infof("prometheus: %v", *prometheus.Spec.Replicas)
+	// check started or not
+	if *prometheus.Spec.Replicas == 0 {
+		logrus.Infof("start prometheus")
+		prometheusCopy := prometheus.DeepCopy()
+		*prometheusCopy.Spec.Replicas = 1
+		if replicas, err := strconv.Atoi(prometheus.Annotations[ReplicaStorageNetworkAnnotation]); err == nil {
+			*prometheusCopy.Spec.Replicas = int32(replicas)
+		}
+		delete(prometheusCopy.Annotations, ReplicaStorageNetworkAnnotation)
+
+		if _, err := h.prometheus.Update(prometheusCopy); err != nil {
+			logrus.Warnf("prometheus update error %v", err)
+			return false
+		}
+		return false
+	}
+
+	return true
+}
+
+func (h *Handler) checkAltermanagerStatusAndStart() bool {
+	// check alertmanager cattle-monitoring-system/rancher-monitoring-alertmanager replica
+	alertmanager, err := h.alertmanagerCache.Get(CattleMonitoringSystemNamespace, RancherMonitoringAlertmanager)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logrus.Infof("Alertmanager not found. skip")
+			return true
+		}
+		logrus.Warnf("alertmanager get error %v", err)
+		return false
+	}
+
+	logrus.Infof("alertmanager: %v", *alertmanager.Spec.Replicas)
+	// check started or not
+	if *alertmanager.Spec.Replicas == 0 {
+		logrus.Infof("start alertmanager")
+		alertmanagerCopy := alertmanager.DeepCopy()
+		*alertmanagerCopy.Spec.Replicas = 1
+		if replicas, err := strconv.Atoi(alertmanager.Annotations[ReplicaStorageNetworkAnnotation]); err == nil {
+			*alertmanagerCopy.Spec.Replicas = int32(replicas)
+		}
+		delete(alertmanagerCopy.Annotations, ReplicaStorageNetworkAnnotation)
+
+		if _, err := h.alertmanager.Update(alertmanagerCopy); err != nil {
+			logrus.Warnf("alertmanager update error %v", err)
+			return false
+		}
+		return false
+	}
+
+	return true
+}
+
+func (h *Handler) checkGrafanaStatusAndStart() bool {
+	// check deployment cattle-monitoring-system/rancher-monitoring-grafana replica
+	grafana, err := h.deploymentCache.Get(CattleMonitoringSystemNamespace, RancherMonitoringGrafana)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logrus.Infof("grafana not found. skip")
+			return true
+		}
+		logrus.Warnf("grafana get error %v", err)
+		return false
+	}
+
+	logrus.Infof("Grafana: %v", *grafana.Spec.Replicas)
+	// check started or not
+	if *grafana.Spec.Replicas == 0 {
+		logrus.Infof("start grafana")
+		grafanaCopy := grafana.DeepCopy()
+		*grafanaCopy.Spec.Replicas = 1
+		if replicas, err := strconv.Atoi(grafana.Annotations[ReplicaStorageNetworkAnnotation]); err == nil {
+			*grafanaCopy.Spec.Replicas = int32(replicas)
+		}
+		delete(grafanaCopy.Annotations, ReplicaStorageNetworkAnnotation)
+
+		if _, err := h.deployments.Update(grafanaCopy); err != nil {
+			logrus.Warnf("Grafana update error %v", err)
+			return false
+		}
+		return false
```

Comment:
similar;  

each successful `Update` still needs to wait 5s for next round to stop/start another POD, it is not reasonable to do so.

simply set `Spec.Replicas` as more as possible,  as long as you set it successful, it's done.  k8s's mechanism will reconcile the remaining work.

Reply:
It will only wait for 5s for all Pods startups.